processing/src/main/java/org/apache/druid/query/DefaultQueryMetrics.java

```diff
@@ -295,12 +304,22 @@ public void identity(String identity)
   }
 
   @Override
-  public void emit(ServiceEmitter emitter)
+  public void emit(final ServiceEmitter emitter)
   {
-    checkModifiedFromOwnerThread();
-    for (Map.Entry<String, Number> metric : metrics.entrySet()) {
-      emitter.emit(builder.build(metric.getKey(), metric.getValue()));
+    final Thread currThread = Thread.currentThread();
+    synchronized (lock) {
+      if (ownerThread == null) {
+        ownerThread = currThread;
+      } else if (!ownerThread.equals(currThread)) {
+        throw new ISE("emit() from multiple threads is not allowed. If it is needed to emit metrics from "
```

Comment:
Not formatted properly. Concatenated string arg should start at it's own line and in the end `);` be on the separate line as well.